lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -121,31 +179,48 @@ def test_query(self, server_config, query_api, user_ok, find_template):
         }
 
         index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
+        # If we're not asking about a particular user, or if the user
+        # field is to be omitted altogether, or if we have a valid
+        # token, then the request should succeed.
+        if not user or user == "no_user" or build_auth_header[1] == "valid":
+            expected_status = HTTPStatus.OK
+        else:
+            expected_status = HTTPStatus.FORBIDDEN
+
         response = query_api(
             "/controllers/list",
             "/_search?ignore_unavailable=true",
             json,
             index,
-            HTTPStatus.OK,
-            status=HTTPStatus.OK,
+            expected_status,
+            headers=build_auth_header[0],
+            status=expected_status,
             json=response_payload,
         )
-        res_json = response.json
-        assert isinstance(res_json, list)
-        assert len(res_json) == 2
-        assert res_json[0]["key"] == "unittest-controller1"
-        assert res_json[0]["controller"] == "unittest-controller1"
-        assert res_json[0]["results"] == 2
-        assert res_json[0]["last_modified_value"] == 1.59847315581e12
-        assert res_json[0]["last_modified_string"] == "2020-08-26T20:19:15.810Z"
-        assert res_json[1]["key"] == "unittest-controller2"
-        assert res_json[1]["controller"] == "unittest-controller2"
-        assert res_json[1]["results"] == 1
-        assert res_json[1]["last_modified_value"] == 1.6
-        assert res_json[1]["last_modified_string"] == "2020-09-26T20:19:15.810Z"
+        assert response.status_code == expected_status
+        if response.status_code == HTTPStatus.OK:
+            res_json = response.json
+            assert isinstance(res_json, list)
+            assert len(res_json) == 2
+            assert res_json[0]["key"] == "unittest-controller1"
+            assert res_json[0]["controller"] == "unittest-controller1"
+            assert res_json[0]["results"] == 2
+            assert res_json[0]["last_modified_value"] == 1.59847315581e12
+            assert res_json[0]["last_modified_string"] == "2020-08-26T20:19:15.810Z"
+            assert res_json[1]["key"] == "unittest-controller2"
+            assert res_json[1]["controller"] == "unittest-controller2"
+            assert res_json[1]["results"] == 1
+            assert res_json[1]["last_modified_value"] == 1.6
+            assert res_json[1]["last_modified_string"] == "2020-09-26T20:19:15.810Z"
 
     def test_empty_query(
-        self, client, server_config, query_api, user_ok, find_template
+        self,
+        client,
+        server_config,
+        query_api,
+        user_ok,
+        find_template,
+        build_auth_header,
     ):
         """
         Check proper handling of a query resulting in no Elasticsearch matches.
```

Comment:
Building the repetition into the `build_auth_header` fixture makes universal documentation a bit awkward: however I like how you did that above, and it might not be a bad idea to add the "three times for ..." here, as well, so anyone debugging is aware of what's happening.